File: JavaScript/Language Features/OOP/src/class.js
// Till ES 5 functions are used to defined a Class in JS. 
// ES6 provides a class contruct to define the class.

function Customer(firstName, lastName) {
    // firstName, lastName are public properties
    this.firstName = firstName;
    this.lastName = lastName;

    this.talk = function() {
        console.log("Hello, my name is " + this.firstName + " " + this.lastName)
    }
}

var cust1 = new Customer("Prasad", "Honrao");
cust1.talk();

var cust2 = new Customer("Jack", "Taylor");
cust2.talk();

File: Java/Language Features/Workspace/Class/src/Class.java
public class Class {
	public static void main(String[] args) {

		Customer newCustomer = new Customer("Prasad", 35);
		System.out.println(newCustomer.name + " " + newCustomer.age);

		Customer anotherCustomer = new Customer();
		anotherCustomer.age = 50;
		anotherCustomer.name = "Jacob";
		System.out.println(anotherCustomer.name + " " + anotherCustomer.age);

	}
}